apps/api/src/services/programs.ts

```diff
@@ -11,7 +14,23 @@ import type { z } from "zod";
 export class ProgramsService {
   constructor(private readonly db: ReturnType<typeof database>) {}
 
-  async getMajors() {
+  async getMajors(query: z.infer<typeof majorQuerySchema>) {
```

Comment:
Why do an early return and not a conditional `where`? It may help to know that a `.where` with no arguments does not add any conditions. Having two code paths with such similar code is an extra maintenance burden and hurts readability, since early return is usually used to checking preconditions or special cases, not a full code path even more complex than the "main" path outside the early return.